### PR TITLE
test: add some coverage

### PR DESCRIPTION
**Pull request by salmanm**
https://github.com/fastify/fastify/issues/2381

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: fastify.js [hunk 1]
```diff
@@ -60,16 +60,16 @@ const onBadUrlContext = {
 }
 
 function defaultErrorHandler (error, request, reply) {
-  if (reply.statusCode >= 500) {
-    reply.log.error(
-      { req: request, res: reply, err: error },
-      error && error.message
-    )
-  } else if (reply.statusCode >= 400) {
+  if (reply.statusCode < 500) {
```
Comment (jsumners) [inline fastify.js:63]: The title of this PR is about improving test coverage. This change does not seem to be related to that goal. Please refrain from doing multiple unrelated things in a single PR.

**Review by jsumners — CHANGES_REQUESTED** [review]

File: fastify.js (hunk 1 above)
Comment (salmanm) [reply fastify.js:63]: It actually is. With the `if` and `else if`, the coverage report expects a case that does not lie in either of the blocks. However, it's not possible here because of [this line](https://github.com/fastify/fastify/blob/master/lib/reply.js#L504). 

So changing it to if/else indeed increases coverage.

Comment (jsumners) [reply fastify.js:63]: I think this is actually different behavior. Is it possible to get an error object in this function with a status code that is not `4xx` or `5xx`.

Comment (salmanm) [reply fastify.js:63]: That's what I tried, but that line I linked above converts all < 400 codes to 500. 
Do you have a way to work around that?

**Review by jsumners — APPROVED** [review]
